#86877cd75 Added end point for salesforce

sab-LC opened the pull request:
**[Issue:](https://app.clickup.com/t/86877cd75)**
  
- Description: Needed an end-point to be accessed by salesforce using the API-KEY and return the user data if exists.

**Solution:**
- Added User serializer
- Updated API urls
- Updated views to return user object based on the query params


**Screen shots**

- postman

<img width="923" alt="image" src="https://github.com/localcontexts/localcontextshub/assets/145371882/02323517-f131-4214-9776-ba81b985ecf2">

<img width="923" alt="image" src="https://github.com/localcontexts/localcontextshub/assets/145371882/50dbfb39-d9ae-474c-b85d-62e6705fe4d4">


- endpoint overview

<img width="923" alt="image" src="https://github.com/localcontexts/localcontextshub/assets/145371882/dcffde9f-ed5f-47f3-833b-2d17d8660444">

arojas1 reviewed (changes requested):
1. Can you remove the endpoint from the home page? Since this should only be accessed by salesforce, it shouldn't be public.
2. Would it be better to have this call under v2 of the API? not necessary but v1 may be phased out as we start putting the API keys up so having salesforce connect using v2 now would be best (we wont have to change the connection later)
3. Also, maybe we should make it where the only key that is allowed is the one from salesforce?

sab-LC commented:
> 1. Can you remove the endpoint from the home page? Since this should only be accessed by salesforce, it shouldn't be public.
> 2. Would it be better to have this call under v2 of the API? not necessary but v1 may be phased out as we start putting the API keys up so having salesforce connect using v2 now would be best (we wont have to change the connection later)
> 3. Also, maybe we should make it where the only key that is allowed is the one from salesforce?

- removed end-point from the homepage
- moved the call from the api
- access to this end-point will be only allowed to the certain users

<img width="828" alt="image" src="https://github.com/localcontexts/localcontextshub/assets/145371882/b2f9aa1b-3bba-47cf-9894-2f7fe0649ef3">
